app/src/androidTest/java/com/rlab/sejima/MUTextFieldTests.java

```diff
@@ -0,0 +1,165 @@
+        // Label's color
+        assertEquals(mMUTextField.getLabelColor(), Color.WHITE);
+
+        // Field
+        assertEquals(mMUTextField.getField(), "");
+        // Field's size
+        assertEquals(mMUTextField.getFieldFontSize(), defaultEditTextSize, 0);
+        // Field's font weight
+        assertEquals(mMUTextField.getFieldFontWeight(), Typeface.NORMAL);
+        // Field's color
+        assertEquals(mMUTextField.getFieldColor(), defaultEditTextFontColor);
+
+        // Placeholder
+        assertEquals(mMUTextField.getPlaceHolderText(), "");
+        // Placeholder color
+        assertEquals(mMUTextField.getPlaceHolderFontColor(), defaultEditTextHintColor);
+        // Underline
+        assertEquals(mMUTextField.getUnderlineColor(), Color.TRANSPARENT);
+
+        // Field's comportment
+        assertFalse(mMUTextField.isSecure());
+        assertTrue(mMUTextField.isEditable());
+        assertFalse(mMUTextField.isActive());
+        assertTrue(mMUTextField.isAutoCorrection());
+        assertTrue(mMUTextField.isReturnKeyAvailable());
+
+        assertEquals(mMUTextField.getKeyboardType(),  InputType.TYPE_NULL);
+        assertNull(mMUTextField.getTFListener());
+
+        assertEquals(mMUTextField.getAlignment(), RelativeLayout.ALIGN_PARENT_START);
+    }
+
+    @Test
+    public void customValues() {
+
+        // Label
+        mMUTextField.setLabel("Custom");
+        assertEquals(mMUTextField.getLabel(), "Custom");
+        // Label's size
+        mMUTextField.setLabelFontSize(124);
+        assertEquals(mMUTextField.getLabelFontSize(), 124 * mScale, 0);
+        // Label's font weight
+        mMUTextField.setLabelFontWeight(Typeface.BOLD);
+        assertEquals(mMUTextField.getLabelFontWeight(), Typeface.BOLD);
+        // Label's color
+        mMUTextField.setLabelColor(Color.RED);
+        assertEquals(mMUTextField.getLabelColor(), Color.RED);
+
+        // Field
+        mMUTextField.setField("CUSTOM FIELD");
+        assertEquals(mMUTextField.getField(), "CUSTOM FIELD");
+        // Field's size
+        mMUTextField.setFieldFontSize(35);
+        assertEquals(mMUTextField.getFieldFontSize(), 35 * mScale, 0);
+        // Field's font weight
+        mMUTextField.setFieldFontWeight(Typeface.BOLD_ITALIC);
+        assertEquals(mMUTextField.getFieldFontWeight(), Typeface.BOLD_ITALIC);
+        // Field's color
+        mMUTextField.setFieldColor(Color.BLUE);
+        assertEquals(mMUTextField.getFieldColor(), Color.BLUE);
+
+        // Placeholder
+        mMUTextField.setPlaceHolderText("Custom placeholder");
+        assertEquals(mMUTextField.getPlaceHolderText(), "Custom placeholder");
+        // Placeholder color
+        mMUTextField.setPlaceHolderFontColor(Color.TRANSPARENT);
+        assertEquals(mMUTextField.getPlaceHolderFontColor(), Color.TRANSPARENT);
+        // Underline
+//        mMUTextField.setUnderlineColor(Color.RED);
+//        assertEquals(mMUTextField.getUnderlineColor(), Color.RED);
```

Comment:
Is underline color not already handle?

Reply:
Underline color is handled.
It has been tested on virtual devices but additional heavy resources have to be mocked so as to test it with JUnit. So i commented it.